### ISSUE TITLE
Fit statistics

### ISSUE THREAD

**Issue by phajy** [fjebaker/SpectralFitting.jl#84]
I wonder if there is an easy way to calculate model-dependent fit statistics such as reduced $\chi^2$ values, for example. $\chi^2_\nu = \chi^2 / \text{d.o.f.}$ where the $\chi^2$ value is derived from the data and the fit (model folded through response), and the number of degree of freedom (d.of.) is equal to $(\text{number of data bins}) - (\text{number of free parameters})$ which come from the data and model, respectively.

**Comment by fjebaker** [comment]
Yes we can add a function that does that. The number of free parameters is already tracked, and the number of data bins is simply `count(mask)` on the data mask, so it's just about adding that to the report card of the fit! Good idea!

**Comment by jamesrafe** [comment]
Good day @phajy and @fjebaker, I have opened a PR for this. If you are available to help review the PR, and help make sure that the math and the usage of variables is correct for the reduced statistics, that would be great! Thank you in advance.
